### PR TITLE
Main branch Edge to Edge implementation

### PR DESCRIPTION
**Pull request by the-scrambler**
Merge pending codelab update.

### CODE REVIEW

**Review by android-dev-lxl — CHANGES_REQUESTED** [review]
- Please remove   `freeCompilerArgs += "-opt-in=androidx.compose.material3.ExperimentalMaterial3Api"`
- Upgrade Gradle to 8.1.0
- Upgrade Compose BOM version 
- Upgrade `kotlinCompilerExtensionVersion`
- Please upgrade other libs like `navigation-compose`

Not sure how we missed this in M3 and DSL upgrades

**Review by android-dev-lxl — APPROVED** [review]

File: app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt [hunk 1]
```diff
@@ -107,16 +109,16 @@ fun LunchTrayApp() {
...
         ) {
             composable(route = LunchTrayScreen.Start.name) {
                 StartOrderScreen(
                     onStartOrderButtonClicked = {
                         navController.navigate(LunchTrayScreen.Entree.name)
                     },
                     modifier = Modifier
-                        .padding(dimensionResource(R.dimen.padding_medium))
                         .fillMaxSize()
+                        .padding(innerPadding)
```
Comment (johnshea) [inline app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt:120]: Combine the two `.padding()`  function calls into 1 or remove 1 depending on what you need.

File: app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt [hunk 2]
```diff
@@ -134,8 +136,9 @@ fun LunchTrayApp() {
                         viewModel.updateEntree(item)
                     },
                     modifier = Modifier
-                        .padding(dimensionResource(R.dimen.padding_medium))
                         .verticalScroll(rememberScrollState())
+                        .padding(innerPadding)
```
Comment (johnshea) [inline app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt:140]: Combine the two `.padding()`  function calls into 1 or remove 1 depending on what you need.

File: app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt [hunk 3]
```diff
@@ -153,8 +156,9 @@ fun LunchTrayApp() {
                         viewModel.updateSideDish(item)
                     },
                     modifier = Modifier
-                        .padding(dimensionResource(R.dimen.padding_medium))
                         .verticalScroll(rememberScrollState())
+                        .padding(innerPadding)
```
Comment (johnshea) [inline app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt:160]: Combine the two `.padding()`  function calls into 1 or remove 1 depending on what you need.

File: app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt [hunk 4]
```diff
@@ -172,8 +176,9 @@ fun LunchTrayApp() {
                         viewModel.updateAccompaniment(item)
                     },
                     modifier = Modifier
-                        .padding(dimensionResource(R.dimen.padding_medium))
                         .verticalScroll(rememberScrollState())
+                        .padding(innerPadding)
```
Comment (johnshea) [inline app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt:180]: Combine the two `.padding()`  function calls into 1 or remove 1 depending on what you need.

File: app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt [hunk 5]
```diff
@@ -189,8 +194,9 @@ fun LunchTrayApp() {
                         navController.popBackStack(LunchTrayScreen.Start.name, inclusive = false)
                     },
                     modifier = Modifier
-                        .padding(dimensionResource(R.dimen.padding_medium))
                         .verticalScroll(rememberScrollState())
+                        .padding(innerPadding)
```
Comment (johnshea) [inline app/src/main/java/com/example/lunchtray/LunchTrayScreen.kt:198]: Combine the two `.padding()`  function calls into 1 or remove 1 depending on what you need.

**Review by johnshea — APPROVED** [review]
LGTM with some small changes.

- Correct `.padding()` issues noted in the file.
- Upgrade gradle wrapper to `8.2`

**Pull request merged**